pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/bucket/BucketDelayedDeliveryTracker.java

```diff
@@ -764,6 +807,10 @@ public synchronized boolean containsMessage(long ledgerId, long entryId) {
                 .orElse(false);
     }
 
+    private String createSnapshotKey(long ledgerId, long entryId) {
+        return ledgerId + ":" + entryId;
+    }
```

Comment:
Instead of a String key, it's better to use a specific record `record SnapshotKey(long ledgerId, long entryId)` since that would avoid overhead of creating a String instance.